### PR TITLE
fix for issue Rendering: incorrect size of circuit box #414

### PR DESCRIPTION
**Pull request by oradomskyi**
Setting recursive=False in the inner loop of circuit rendering seems to fix the issue, txt pdf and html files are generated correctly and contain the correct circuit of proper size.

image

files generated for the circuit mentioned in the original issue are here:
https://drive.google.com/drive/folders/1E1cF_RRrfD8HEcfvpSkX1at810IYkHhl?usp=sharing

### CODE REVIEW

**Comment by oradomskyi** [comment]
image

image

**Comment by oradomskyi** [comment]
PR is related to the box size, not the positioning of heralds - that is another issue

**Comment by raksharuia** [comment]
Yes, fixing of heralds (changes have already been made for this) is another issue; your fix corrects the issue of incorrect box size on rendering. Thanks for the contribution :-)

File: perceval/rendering/circuit/renderer.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def render_circuit(self,
                     if c.is_composite() and c._components:
                         if recursive:
                             self._current_subblock_info = self._subblock_info.setdefault(c, {})
-                            self.open_subblock(shiftr, c.name, self.get_circuit_size(c, recursive=True), c._color)
+                            self.open_subblock(shiftr, c.name, self.get_circuit_size(c, recursive=False), c._color)
                             self.render_circuit(
```
Comment (raksharuia) [inline perceval/rendering/circuit/renderer.py:108]: Could you please add a unittest (following our conventions on testing circuit images) using the example from the issue?

**Review by raksharuia — CHANGES_REQUESTED** [review]

File: perceval/rendering/circuit/renderer.py (hunk 1 above)
Comment (oradomskyi) [reply perceval/rendering/circuit/renderer.py:108]: This is done, let me know if you would like to see other circuit tests.

P.S. This issue seems to be specific to Canvas Renderer. Text Renderer appears to have no problems with box size.

Comment (oradomskyi) [reply perceval/rendering/circuit/renderer.py:108]: Failing tests on Windows does not seem to be related to the proposed changes.

**Pull request merged**